src/sensors/include/sensors/imu _subscriber.hpp

```diff
@@ -0,0 +1,30 @@
+#include <memory>
+#include <fstream>
+#include "rclcpp/rclcpp.hpp"
+#include "sensor_msgs/msg/imu.hpp"
+
+class IMUSubscriber : public rclcpp::Node
+{
+public:
+    IMUSubscriber(const std::string &csv_file_path);
+        
+    
+    ~IMUSubscriber();
+    
+
+private:
+    void imu_callback(const sensor_msgs::msg::Imu::SharedPtr msg);
+
+    std::ofstream csv_file_;
+    std::string csv_file_path_;
+    rclcpp::Subscription<sensor_msgs::msg::Imu>::SharedPtr subscription_;
+
+    // Calibration offsets
+    double gyro_x_offset_;
```

Comment:
The naming convention for naming class member variables is as follows:
1. all member variable name must start with the character m. For example gyro_x_offset_ must be mpGyroXOffset
2. to specify that the variable name is private use the character p. For example gyro_x_offset is private so it's name is mpGyroXOffset